src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -207,10 +207,65 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
                 }
             }
 
+            // We may get a flood of diagnostic notificaitons on the BG.  In order to not overwhelm
+            // the UI thread with work to do, and to avoid excess allocations, we batch up these
+            // notifications, and attempt to process them all at once on the UI thread every 50ms
+            // or so.
+            //
+            // Because we're batching, we can optimize things such that we only store two pieces
+            // of data per provider.  Specifically, if they were removing all diagnostics, and
+            // then the last diagnostics they wanted to create.
```

Comment:
note: it may be possible to optimize things even further.  i.e. just store the last DiagnosticsUpdateArgs for a provider.  However, i wasn't sure if that would end up breaking anything.  i.e. it's not clear to me if there are circumstances where we depend on processing the 'remove' before the 'create'.  So i wanted to make the safest change possible.